posts/2020-11-20-JNDI-gRPC-200012.adoc

```diff
@@ -0,0 +1,389 @@
+
+
+[source, XML]
+----
+   <ejbContainer bindToJavaGlobal="false"/>
+----
+
+Finally, the following new configuration attribute on the `<ejbContainer>` element in open-liberty enables the failing application start when multiple beans are bound to the same JNDI name:
+
+[source, XML]
+----
+    <ejbContainer customBindingsOnError="FAIL"/>
+----
+
+[#java15]
+== Support for Java SE 15
+
+Any official Java SE 15 release from link:https://adoptopenjdk.net?variant=openjdk15&jvmVariant=openj9[AdoptOpenJDK], link:https://jdk.java.net/15/[Oracle], or other OpenJDK vendor will work with Open Liberty. Java SE 15 is not a long-term supported release, with standard support scheduled to end in March 2021.
+
+Keep in mind, Eclipse OpenJ9 link:{url-prefix}/blog/2019/10/30/faster-startup-open-liberty.html[typically offers faster startup times] than Hotspot.
+
+The primary features added in this release include:
+
+* link:https://openjdk.java.net/jeps/379[JEP 379] Shenandoah: A Low-Pause-Time Garbage Collector
+* link:https://openjdk.java.net/jeps/377[JEP 377] ZGC: A Scalable Low-Latency Garbage Collector
+* link:https://openjdk.java.net/jeps/378[JEP 378] Text Blocks
+* link:https://openjdk.java.net/jeps/384[JEP 384] Records (Second Preview)
+* link:https://openjdk.java.net/jeps/360[JEP 360] Sealed Classes (Preview)
+
+For more information on downloading a version of Java 15, see link:https://adoptopenjdk.net/index.html?variant=openjdk15&jvmVariant=openj9[AdoptOpenJDK.net], link:https://www.eclipse.org/openj9/[Eclipse.org] or link:https://openjdk.java.net/groups/hotspot[OpenJDK.java.net].
+
+For working with the `server.env` file in Open Liberty, see the `Configuration Files` section of the Open Liberty link:{url-prefix}/docs/latest/reference/config/server-configuration-overview.html[Server Configuration Overview documentation].
+
+For more information on new features available in Java 15, see link:https://openjdk.java.net/projects/jdk/15/[OpenJDK].
+
+[#bugs]
+== Significant bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3A%22release+bug%22+label%3Arelease%3A200012[full list of bugs fixed in 20.0.0.12].
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14542[IllegalAccessError when using MP Rest Client with Java 15]
++
+There was a MP Rest Client failure when running with Java 15. This was fixed by changing the visibility of the `compareCustomStatus` method to public. If you would like to know more about MP Rest Client check out our link:https://openliberty.io/guides/microprofile-rest-client.html[Consuming RESTful services with template interfaces] and link:https://openliberty.io/guides/microprofile-rest-client-async.html[Consuming RESTful services asynchronously with template interfaces] guides.
```

Comment:
"... with Java SE 15."

Reply:
@andymc12 how was this a release bug if we didn't support Java SE 15 on OL prior to 20.0.0.12?

Reply:
Done